### PR TITLE
sorted constant arrays and moved classes into alphabetical order

### PR DESCRIPTION
**Pull request by RomanGoEmpire**
### TL;DR

Reorganized constants and enums, and updated JSON typing structure.

### What changed?

- Alphabetically sorted file type lists in `constants.py`
- Reordered enum classes in `enums.py`
- Restructured JSON typing in `json_typing.py`, moving `JsonLibary` to the bottom and change type hints

* **#2** <a href="https://app.graphite.dev/github/pr/RomanGoEmpire/TagStudio/2?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1** <a href="https://app.graphite.dev/github/pr/RomanGoEmpire/TagStudio/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

### CODE REVIEW

**Comment by RomanGoEmpire** [comment]
* **#1** <a href="https://app.graphite.dev/github/pr/RomanGoEmpire/TagStudio/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RomanGoEmpire and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
